### ISSUE TITLE
Unable to disable nor choose which signs to enable

### ISSUE THREAD

**Issue by gonviegas** [dstein64/nvim-scrollview#99]
```signs_on_startup``` seems to not be working at all.
Im on the lastest neovim release using lazy. 
This is my nvim-scrollview setup:

```
return {
  'dstein64/nvim-scrollview',
  config = function()
    require("scrollview").setup({
      excluded_filetypes = { 'NvimTree' },
      current_only = true,
      column = 1,
      signs_on_startup = {},
    })
  end
}
```

**Comment by dstein64** [comment]
@gonviegas, is the issue resolved after 3138714544ad7b9c1cab77dc74238624dd16d9b9?

This sounds like the same [issue](https://github.com/dstein64/nvim-scrollview/issues/98#issuecomment-1585147610) encountered by @Fausto-Korpsvart.

Some of the functionality was being initialized asynchronously to work around [Neovim #13720](https://github.com/neovim/neovim/issues/13720). I switched back to synchronous initialization in 23c4e10f5b84f33a4b3992d3ef8c16d4863f5ed1, since [Neovim #13720](https://github.com/neovim/neovim/issues/13720) was resolved.

However, I'll keep the existing behavior if it prevents other issues, like the ones that you and @Fausto-Korpsvart are encountering. Unfortunately, a downside of asynchronous initialization is that some of the plugin's events will be omitted from the output when using `nvim --startuptime`, which could complicate profiling.

**Comment by Fausto-Korpsvart** [comment]
> @gonviegas, is the issue resolved after [3138714](https://github.com/dstein64/nvim-scrollview/commit/3138714544ad7b9c1cab77dc74238624dd16d9b9)?
> 
> This sounds like the same [issue](https://github.com/dstein64/nvim-scrollview/issues/98#issuecomment-1585147610) encountered by @Fausto-Korpsvart.
> 
> Some of the plugin's functionality was being initialized asynchronously to work around [Neovim #13720](https://github.com/neovim/neovim/issues/13720). I switched back to synchronous initialization in [23c4e10](https://github.com/dstein64/nvim-scrollview/commit/23c4e10f5b84f33a4b3992d3ef8c16d4863f5ed1), since [Neovim #13720](https://github.com/neovim/neovim/issues/13720) was resolved.
> 
> However, I'll keep the existing behavior if it prevents other issues, like the ones that you and @Fausto-Korpsvart are encountering. Unfortunately, a downside of asynchronous initialization is that some of the plugin's events will be omitted from the output when using `nvim --startuptime`, which could complicate profiling.

@dstein64, I updated a moment ago and everything works normally again, even the folding icons are back too, I went back to the old configuration and everything works fine.
Much appreciated. }:-]

**Comment by gonviegas** [comment]
> @gonviegas, is the issue resolved after [3138714](https://github.com/dstein64/nvim-scrollview/commit/3138714544ad7b9c1cab77dc74238624dd16d9b9)?
> 
> This sounds like the same [issue](https://github.com/dstein64/nvim-scrollview/issues/98#issuecomment-1585147610) encountered by @Fausto-Korpsvart.
> 
> Some of the plugin's functionality was being initialized asynchronously to work around [Neovim #13720](https://github.com/neovim/neovim/issues/13720). I switched back to synchronous initialization in [23c4e10](https://github.com/dstein64/nvim-scrollview/commit/23c4e10f5b84f33a4b3992d3ef8c16d4863f5ed1), since [Neovim #13720](https://github.com/neovim/neovim/issues/13720) was resolved.
> 
> However, I'll keep the existing behavior if it prevents other issues, like the ones that you and @Fausto-Korpsvart are encountering. Unfortunately, a downside of asynchronous initialization is that some of the plugin's events will be omitted from the output when using `nvim --startuptime`, which could complicate profiling.

Yes, it's all working fine now.
Thank you!

**Comment by dstein64** [comment]
I think I know what the issue was.

`nvim-scrollview` was implemented as a traditional Vim-style plugin, in VimScript.

At some point, it was [ported](https://github.com/dstein64/nvim-scrollview/pull/47) to Lua, but the interface remained the same. It was still effectively a Vim-style plugin, using a `plugin/` script for initialization, not a `setup()` function.

There was a request to transition to a [Lua-style](https://github.com/dstein64/nvim-scrollview/issues/69) plugin. That would have broken the plugin for existing users and complicated configuration for users using `init.vim` instead of `init.lua`. I added a Lua `setup()` function for convenience, which effectively set the existing scrollview configuration global variables.

When configuring plugins with `lazy.nvim`, using something like the example above (repeated here) assumes the plugin is not a Vim-style plugin.

```lua
{
  'dstein64/nvim-scrollview',
  config = function()
    require("scrollview").setup({
      excluded_filetypes = { 'NvimTree' },
      current_only = true,
      column = 1,
      signs_on_startup = {},
    })
  end
}
```

The `plugin/` scripts are executed prior to the config function. For the `nvim-scrollview` functionality that had to be set before plugin loading, this was problematic since the `plugin/` script would execute but the configuration had not occurred yet.

A workaround would be to specify an `init` function for `lazy.nvim` instead of a `config` function (haven't tested). It looks like that is the approach recommended for traditional Vim-style plugins.

```lua
{
  'dstein64/nvim-scrollview',
  init = function()
    require("scrollview").setup({
      excluded_filetypes = { 'NvimTree' },
      current_only = true,
      column = 1,
      signs_on_startup = {},
    })
  end
}

-- Or:

{
  'dstein64/nvim-scrollview',
  init = function()
    vim.g.excluded_filetypes = { 'NvimTree' },
    vim.g.current_only = true,
    vim.g.column = 1,
    vim.g.signs_on_startup = {},
  end
}
```

Asynchronous configuration fixed the issue since it deferred the execution of some `nvim-scrollview` initialization until after the `lazy.nvim` `config` function ran. I will keep that approach, since it resolves the issue and won't require users to change from using `config` to `init`. However, a downside is that events will be omitted from `nvim --startuptime`'s output. But events were already being omitted, since the same type of workaround had been used, until recently, to resolve another [issue](https://github.com/neovim/neovim/issues/13720).
